metricbeat/module/redis/key/_meta/docs.asciidoc

```diff
@@ -0,0 +1,25 @@
+The Redis `key` metricset collects information about Redis keys.
+
+For each key matching one of the configured patterns, an event is sent to
+elasticsearch with information about this key, what includes the type, its
+length when available, the value when it is a string, and its ttl.
+
+Patterns are configured as a list containing these fields:
+* `pattern` (required): pattern for key names, as accepted by the Redis
+  `KEYS` or `SCAN` commands.
+* `max` (optional): safeguard when using patterns with wildcards to avoid
+  collecting too many keys (Default: 0, no limit)
+* `keyspace` (optional): Identifier of the database to use to look for the keys
+  (Default: 0)
+
+For example the following configuration will collect information about all keys
+whose name starts with `pipeline-*`, with a limit of 20 keys.
+
+[source,yaml]
+------------------------------------------------------------------------------
+- module: redis
+  metricsets: ['key']
+  key.patterns:
+    - name: 'pipeline-*'
+      max: 20
```

Comment:
should we call this limit instead of max as you describe it as limit above?

Reply:
Yes, maybe this is a better name, I'll rename it.

Reply:
Renamed.